tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -217,20 +239,33 @@ export class GraphExecutor implements FunctionExecutor {
       this.compiledMap.set(compilationKey, orderedNodes);
     }
 
+    try {
+      const keepTensorForDump = env().getBool('KEEP_INTERMEDIATE_TENSORS');
+      if (keepTensorForDump) {
+        this.dumpMode = DumpMode.Sync;
+      }
+    } catch (e) {
+      console.warn(e.message);
+    }
     const tensorArrayMap: TensorArrayMap = {};
     const tensorListMap: TensorListMap = {};
+    const tensorsMap: NamedTensorsMap = {...this.weightMap};
 
-    return tidy(() => {
+    const result = tidy(() => {
       const context = new ExecutionContext(
           this.weightMap, tensorArrayMap, tensorListMap,
           this.functionExecutorMap);
-      const tensorsMap: NamedTensorsMap = {...this.weightMap};
+      if (this.dumpMode === DumpMode.Sync) {
+        this.tensorsPendingDisposal = [];
+      }
 
       Object.keys(inputs).forEach(name => {
         const [nodeName, index] = parseNodeName(name);
         const tensors: Tensor[] = [];
         tensors[index] = inputs[name];
         tensorsMap[nodeName] = tensors;
+        // Input tensors should be disposed by user.
+        this.keepTensors(tensors);
```

Comment:
Why we need to add `keepTensors` for inputs compared with the original code?

Reply:
I add some comments above this line:
```
        // For some models, such as bodypix, it will dispose the input tensors
        // in its top level tidy. In dump mode, these tensors are required, so
        // call keep to eusure they are preserved. However, this comes with a
        // side effect in dump mode, tensor leak.
```

Reply:
This leak only happens in dump mode.  And 'tensors' includes some tensors will be used in later inference, especially in e2e, which will run inference on two different backends.

Reply:
Anyway, I think we need to fix it. We need to find which place releases it and debug why it's released and then reused again in debug mode. Or give reason that why tensor leak is unavoidable.